cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/upgradesstable/UpgradeSSTableContext.java

```diff
@@ -0,0 +1,166 @@
+            };
+
+    /**
+     * Creates a new UpgradeSSTableContext
+     * @param nodes The nodes that will be have their SSTables upgraded.
+     * @param keySpaces The key spaces whose SSTables will be upgraded. If empty, all
+     *                  non-system key spaces will be cleaned up.
+     * @param columnFamilies The column families whose SSTables will be upgraded. If
+     *                       empty, all column families will be clean up.
+     * @return A UpgradeSSTableContext constructed from the parameters.
+     */
+    @JsonCreator
+    public static UpgradeSSTableContext create(
+            @JsonProperty("nodes") final List<String> nodes,
+            @JsonProperty("key_spaces") final List<String> keySpaces,
+            @JsonProperty("column_families") final List<String> columnFamilies) {
+        return new UpgradeSSTableContext(nodes, keySpaces, columnFamilies);
+    }
+
+    @JsonProperty("nodes")
+    private final List<String> nodes;
+    @JsonProperty("key_spaces")
+    private final List<String> keySpaces;
+    @JsonProperty("column_families")
+    private final List<String> columnFamilies;
+
+    /**
+     * Constructs a new UpgradeSSTableContext
+     * @param nodes The nodes that will be have their SSTables upgraded.
+     * @param keySpaces The key spaces whose SSTables will be upgraded. If empty, all
+     *                  non-system key spaces will be cleaned up.
+     * @param columnFamilies The column families whose SSTables will be upgraded. If
+     *                       empty, all column families will be clean up.
+     */
+    public UpgradeSSTableContext(final List<String> nodes,
+                                 final List<String> keySpaces,
+                                 final List<String> columnFamilies) {
+        this.nodes = (nodes == null) ? Collections.emptyList() : nodes;
+        this.keySpaces = (keySpaces == null) ?
+                Collections.emptyList() :
+                keySpaces;
+        this.columnFamilies = (columnFamilies == null) ?
+                Collections.emptyList() :
+                columnFamilies;
+    }
+
+    /**
+     * Gets the nodes for the SSTable upgrade.
+     * @return The names of the nodes that will be upgraded.
+     */
+    @JsonIgnore
+    public List<String> getNodes() {
+        return nodes;
+    }
+
+    /**
+     * Gets the column families.
+     * @return The column families that will be upgraded. If empty, all column
+     * families will be upgradeed.
+     */
+    @JsonIgnore
+    public List<String> getColumnFamilies() {
+        return columnFamilies;
+    }
+
+    /**
+     * Get the key spaces that will be upgraded.
+     * @return The key spaces that will be upgraded. If empty, all non-system
+     * key spaces will be upgraded.
+     */
+    @JsonIgnore
+    public List<String> getKeySpaces() {
+        return keySpaces;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
This standard pattern is followed almost in every class.

Null check is not required because the next line `if (!(o instanceof UpgradeSSTableContext)) return false;` returns false if o is null.